### PR TITLE
[Review Ready]Wing mutant body parts and Angel wings

### PR DESCRIPTION
**Pull request by Shadowlight213**
Currently I only have the basic wing mutant body parts done.

Angel wings are going to be used for a future Archangel antag
The archangel is sort of going to be the divine judgement against the sintouched.
Sort of an opposite to the lawyer demon.
Angel wings are going to be admin/archangelonly however roundstart wings is code supported in case medbay wants to be ~~succubi~~ saviors.

I'm mainly working on the wing component

Update: I've refactored the Angel flight abilities to be based on a species. This should make things way less hacky.
Progress:
Angel wings themselves will offer a number of advantages and disadvantages:
- [x] Angel wings act like jetpacks while flying in an atmosphere
- [x] Being stunned while flying causes you to lose control and be lube launched and stunned for a long time

~~\- [ ] True demon can completely rip an angel's wings off if they get them stunned.~~ This is going to need more sprites to do, so it will have to be done in a later PR.

Wings general:
- [x] Actually get sprites that aren't the griffon cloak
- [x] Rework mutant body parts to allow for flagging parts as restricted from roundstart

Code cleanup:
- [x] See if it's viable to datumize the flight system.
- [x] Change everything to a species so it's a million times cleaner.

EXPERIMENTAL:
~~\- [ ] See if it's possible to make it an organ.~~
This is going to need more sprites to do, so it will have to be done in a later PR.

Kickass sprites done by Kor's GF.

wings

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
My god, it's what the mutant bodypart system was allegedly for!

**Comment by KorPhaeron** [comment]
>however roundstart wings is code supported in case medbay wants to be succubi saviors.

@Iamgoofball was right

**Comment by KazeEspada** [comment]
Yay more customization options for the most racist server.

**Comment by FantasticFwoosh** [comment]
Pass the redbull over here matey. It's given us wings.

**Comment by xxalpha** [comment]
>wings

uhhhhh

**Comment by Core0verload** [comment]
Mutant parts really, really need a rewrite, :+1: 

However, eventually this will result in roundstart angel catgirls.

**Comment by TechnoAlchemisto** [comment]
t-they'll never be roundstart, right?

**Comment by Iamgoofball** [comment]
# SLIPPERY

#SLOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOÒOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOPE

**Comment by Shadowlight213** [comment]
Archangel wings will definitely be admin only. They provide too many advantages to be roundstart.
Cosmetic wings..probably not roundstart.
Remember, open wings are going to drastically increase your hitbox for melee combat. It's a mechanical nerf to play with them.

**Comment by TechnoAlchemisto** [comment]
so are cat ears and lizard tails

**Comment by TrustyGun** [comment]
we hippiestation now

**Comment by Jazaen** [comment]
>Archangel antag
Hans... Are we the baddies?

**Comment by octareenroon91** [comment]
>holy antag

It's a valids/faction issue.

Angels that happen to be on the station do not share the same contractual relationship of trust that binds all crewmembers together.

Centcom roles might also be called 'antag' for that reason.



Shadowlight, do you think you can fit in fly wings?

**Comment by Jazaen** [comment]
Not really. "Antag" is a thing that works against the crew. Thus, most centcom roles are not antags, with the exception of Deathsquid, and even then some DS teams actually RESCUE crewmembers (weird, right?)

Anyway, since there is lots of murder, cruel medical experimentation, slavery and total lack of ethics on SS13, sometimes all at once, I wouldn't be surprised if we actually were the bad guys. At least by comparison. Kind of like some of the 40K factions - evil by our standards, but meh by theirs.

**Comment by Shadowlight213** [comment]
That's sort of the point behind it. You better be careful about stealing those yellow gloves, while the archangel is around or they might just holy judgement your ass.

@octareenroon91 give me sprites, and I can. The wings are just as easy to add onto as the ears/tail. The hard part is adding the new bodypart itself. Adding new options is extremely simple codewise.

**Comment by KorPhaeron** [comment]
>assuring us it wont be roundstart while arguing it wont be a balance issue for it to be roundstart

You can't fool me

**Comment by Shadowlight213** [comment]
@KorPhaeron angel wings specifically are NEVER going to be round start. Ever. No maintainer in thwir right mind would allow it. Im not even going to make ot a config. They are far too effective.
There is no balance issue for non angel wings at all though, because as I've said, the flying is angel only. All other wings would be cosmetic with a downside of a larger hitbox.
Although hitbox depends on the actual wing sprite.

**Comment by PKPenguin321** [comment]
Just make a wings jetpack for your antag instead of enabling bird fetishists please

**Comment by Shadowlight213** [comment]
I can comment out the wings from roundstart selection completely, rather than make it a config if it's triggering people too much.
@pkpenguin321 making a jetpack would be relatively simple, true. However this is partially a test on my half to see what I can accomplish with the mutant parts system.

**Comment by PKPenguin321** [comment]
suuuure it is

**Comment by KorPhaeron** [comment]
@Shadowlight213 

So does this stuff work?

**Comment by KorPhaeron** [comment]
Also I feel like this should probably take up your back slot somehow. Maybe add the species flag slot blocking stuff or something

**Comment by Shadowlight213** [comment]
@KorPhaeron yes, this actually does work.
I've tested the stun effect and the jetpack effect and they both work.
I still need to fix up the float effect as update gravity is  causing the float to sort of stutter as it tries to switch it off while the species code is trying to turn it on.

**Comment by Jazaen** [comment]
Make wings a bodypart, and add wing transplantation surgery! BECAUSE KILLING SERVANTS OF HIGHER POWERS AND GRAFTING PARTS OF THEIR BODiES SEEMS LIKE SOMETHING NT WOULD DO.

**Comment by Shadowlight213** [comment]
@Jazaen No. because doing that would require me to literally throw out 80% of my work and start over from scratch.

**Comment by Shadowlight213** [comment]
Ran a testmerge round.
TODO reminders list
Add wings giving protection from chasms (laval?), wet floors
Make the no backpack restriction be removed if wearing a hardsuit
Add a downside/reason to not fly nonstop

**Comment by Supermichael777** [comment]
stamina damage

**Comment by Shadowlight213** [comment]
Stamina damage might work. Probably stamina damage gained by moving, with damage healing slowed while hovering.

**Comment by KorPhaeron** [comment]
Is a cooldown not possible? Sprinting+self knockout was terrible

**Comment by Shadowlight213** [comment]
@KorPhaeron the stamina damage would be slower than sprinting.
Also I can make the flying deactivate shortly before knockout level.

**Comment by KorPhaeron** [comment]
Finish this I wanna fly around

**Comment by Shadowlight213** [comment]
Ok, I decided not to include the stamina damage. This should be ready to merge i guess.

**Comment by optimumtact** [comment]
@Shadowlight213 has this had a test merge?

Can you arrange for one for 24 hours or something?

**Comment by KorPhaeron** [comment]
Not a recent test merge no

**Comment by Shadowlight213** [comment]
yeah. sure

**Comment by KorPhaeron** [comment]
Is there any reason why an addwings button like purrbation wouldnt work?

**Comment by Shadowlight213** [comment]
@KorPhaeron It being a species allows me to have all the flying code localized to the species datum.

**Comment by Shadowlight213** [comment]
Also I think I broke cat ears somewhow. I need to look into this

**Comment by optimumtact** [comment]
Fix the conflicts and I'm fairly happy to pull the trigger on this

**Comment by Shadowlight213** [comment]
@optimumtact done

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -23,24 +23,25 @@
...
+#define ID_LAYER				19
+#define SHOES_LAYER				18
+#define GLOVES_LAYER			17
+#define SUIT_LAYER				16
+#define BELT_LAYER				15		//Possible make this an overlay of somethign required to wear a belt?
+#define SUIT_STORE_LAYER		14
+#define BODY_WING_LAYER			13		//Optimum layer for wing mutant parts. below hair and headgear, and above shit on the body. Can't do better without directional layers
+#define BACK_LAYER				12
+#define EARS_LAYER				11
+#define GLASSES_LAYER			10
```
Comment (phil235) [inline code/__DEFINES/misc.dm]: why the changes here? I mean outside of adding the wing layer.

Comment (Shadowlight213) [inline code/__DEFINES/misc.dm]: Most of the other changes were done to make it look nicer. It's hard to explain it just codewise, but I moved all the stuff around until it looked more logical in game.

**Comment by Cheridan** [comment]
the slope has been lubed

**Comment by HankMoodyExperience** [comment]
>NOT GIVING THE LAWYER WINGS ROUNDSTART AND RENAMING HIM HARVEY BIRDMAN: ATTORNEY AT LAW

REFERENCE MISSED

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 2]
```diff
@@ -725,3 +725,148 @@ SYNDICATE BLACK OPS
...
+		H << "<span class='warning'>The atmosphere is too thin for you to fly!</span>"
+		return 0
+	else
+		return 1
+
+/datum/action/innate/flight
+	name = "Toggle Flight"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUNNED
+	button_icon_state = "slimesplit"
+	background_icon_state = "bg_alien"
```
Comment (Incoming5643) [inline code/modules/mob/living/carbon/human/species_types.dm]: Quack

File: code/__DEFINES/misc.dm [hunk 3]
```diff
@@ -23,24 +23,25 @@
...
+#define FRONT_MUTATIONS_LAYER	22		//mutations that should appear above body, body_adj and bodyparts layer (e.g. laser eyes)
+#define DAMAGE_LAYER			21		//damage indicators (cuts and burns)
+#define UNIFORM_LAYER			20
+#define ID_LAYER				19
+#define SHOES_LAYER				18
+#define GLOVES_LAYER			17
+#define SUIT_LAYER				16
+#define BELT_LAYER				15		//Possible make this an overlay of somethign required to wear a belt?
+#define SUIT_STORE_LAYER		14
+#define BODY_WING_LAYER			13		//Optimum layer for wing mutant parts. below hair and headgear, and above shit on the body. Can't do better without directional layers
```
Comment (Incoming5643) [inline code/__DEFINES/misc.dm]: BODY_BEHIND_LAYER
BODY_ADJ_LAYER
FRONT_MUTATIONS_LAYER

ARE THE DIRECTIONAL LAYERS

Comment (Shadowlight213) [inline code/__DEFINES/misc.dm]: I tried that. Uniform was rendered above the wings when facing north. Honk honk.
